Sources/Swift/Integrations/SessionReplay/SentryReplayOptions.swift

```diff
@@ -152,103 +207,206 @@ public class SentryReplayOptions: NSObject, SentryRedactOptions {
+    /// - Parameter dictionary: A dictionary containing the configuration options for the session replay.
+    ///
+    /// - Warning: This initializer is primarily used by Hybrid SDKs and is not intended for public use.
     convenience init(dictionary: [String: Any]) {
-        let sessionSampleRate = (dictionary["sessionSampleRate"] as? NSNumber)?.floatValue ?? 0
-        let onErrorSampleRate = (dictionary["errorSampleRate"] as? NSNumber)?.floatValue ?? 0
-        let maskAllText = (dictionary["maskAllText"] as? NSNumber)?.boolValue ?? true
-        let maskAllImages = (dictionary["maskAllImages"] as? NSNumber)?.boolValue ?? true
-        let enableViewRendererV2 = (dictionary["enableViewRendererV2"] as? NSNumber)?.boolValue ?? (dictionary["enableExperimentalViewRenderer"] as? NSNumber)?.boolValue ?? false
-        let enableFastViewRendering = (dictionary["enableFastViewRendering"] as? NSNumber)?.boolValue ?? false
+        // This initalizer is calling the one with optional parameters, so that defaults can be applied
+        // for absent values.
+        self.init(
+            sessionSampleRate: (dictionary["sessionSampleRate"] as? NSNumber)?.floatValue,
+            onErrorSampleRate: (dictionary["errorSampleRate"] as? NSNumber)?.floatValue,
+            maskAllText: (dictionary["maskAllText"] as? NSNumber)?.boolValue,
+            maskAllImages: (dictionary["maskAllImages"] as? NSNumber)?.boolValue,
+            enableViewRendererV2: (dictionary["enableViewRendererV2"] as? NSNumber)?.boolValue
+            ?? (dictionary["enableExperimentalViewRenderer"] as? NSNumber)?.boolValue,
+            enableFastViewRendering: (dictionary["enableFastViewRendering"] as? NSNumber)?.boolValue,
+            maskedViewClasses: (dictionary["maskedViewClasses"] as? NSArray)?.compactMap({ element in
+                NSClassFromString((element as? String) ?? "")
+            }),
+            unmaskedViewClasses: (dictionary["unmaskedViewClasses"] as? NSArray)?.compactMap({ element in
+                NSClassFromString((element as? String) ?? "")
+            }),
+            quality: SentryReplayQuality.from(rawValue: dictionary["quality"] as? Int),
+            sdkInfo: dictionary["sdkInfo"] as? [String: Any],
+            frameRate: (dictionary["frameRate"] as? NSNumber)?.uintValue,
+            errorReplayDuration: (dictionary["errorReplayDuration"] as? NSNumber)?.doubleValue,
+            sessionSegmentDuration: (dictionary["sessionSegmentDuration"] as? NSNumber)?.doubleValue,
+            maximumDuration: (dictionary["maximumDuration"] as? NSNumber)?.doubleValue
+        )
+    }
+
+    /**
+     * Initializes a new instance of ``SentryReplayOptions`` with the specified parameters.
+     *
+     * - Parameters:
+     *   - sessionSampleRate: Indicates the percentages in which replay for the session will be created.
+     *   - onErrorSampleRate: Indicates the percentages in which a buffer of 30 seconds replay will be send with error events.
```

Comment:
`m`: I'm a bit confused by the wording here. I don't really understand this. Maybe try to rephrase it, please.

Reply:
I agree, these were already in the class before I started refactoring. Will replace them.

Reply:
> these were already in the class before I started refactoring

That explains it, then 😅 .  I'm used to higher standard comments from you 😄

Reply:
I rewrote them, should be more concise now.